metriq_gym/benchmarks/lr_qaoa.py

```diff
@@ -0,0 +1,440 @@
+    """
+
+    max_cost = cost_maxcut(optimal, graph)
+    optimal_probability = 0.0
+    total_cost = 0.0
+    shots = 0.0
+    for bitstring, counts in samples_dict.items():
+        cost = cost_maxcut(bitstring, graph)
+        total_cost += counts * cost
+        if math.isclose(cost, max_cost):
+            optimal_probability += counts
+
+        if cost > max_cost:
+            print(f"There is a better cost than that of Simulated Annealing: {cost - max_cost}")
+        shots += counts
+    approx_ratio = total_cost / (max_cost * shots)
+    optimal_probability /= shots
+    return {"approx_ratio": approx_ratio, "optimal_probability": optimal_probability}
+
+
+def random_samples(num_samples: int, n_qubits: int) -> dict:
+    """
+    Generates random bitstring samples for a given number of qubits.
+
+    Parameters:
+    num_samples (int): The number of random bitstrings to generate.
+    n_qubits (int): The number of qubits (length of each bitstring).
+
+    Returns:
+    dict: A dictionary where keys are randomly generated bitstrings
+          and values are their occurrence counts.
+    """
+
+    random_samples = {}
+
+    for _ in range(num_samples):
+        bitstring = "".join(random.choice(["0", "1"]) for _ in range(n_qubits))
+        if bitstring not in random_samples:
+            random_samples[bitstring] = 0
+        random_samples[bitstring] += 1
+
+    return random_samples
+
+
+def calc_random_stats(num_qubits, graph_info, shots, num_random_trials, optimal_sol):
+    graph = nx.Graph()
+    graph.add_nodes_from(range(num_qubits))
+    graph.add_weighted_edges_from(graph_info)
+    approx_ratio_random_list = []
+    for _ in range(num_random_trials):
+        random_samples_dict = random_samples(shots, len(graph.nodes()))
+        approx_ratio_random_list.append(
+            objective_func(random_samples_dict, graph, optimal_sol)["approx_ratio"]
+        )
+    approx_ratio_random_mean = statistics.mean(approx_ratio_random_list)
+    approx_ratio_random_std = statistics.pstdev(approx_ratio_random_list)
+    return approx_ratio_random_mean, approx_ratio_random_std
+
+
+@dataclass
+class LinearRampQAOAData(BenchmarkData):
+    provider_job_ids: list[str]
+    num_qubits: int
+    graph_info: list[list]
+    graph_type: GraphType
+    optimal_sol: str
+    trials: int
+    num_random_trials: int
+    confidence_level: float
+    shots: int
+    qaoa_layers: list[int]
+    seed: int
+    delta_beta: float
+    delta_gamma: float
+    approx_ratio_random_mean: float
+    approx_ratio_random_std: float
+    circuit_encoding: EncodingType
+
+
+class LinearRampQAOAResult(BenchmarkResult):
+    num_qubits: int
+    qaoa_layers: list[int]
+    approx_ratio: list[float]
+    random_approx_ratio: float
+    optimal_probability: list[float]
+    confidence_pass: list[bool]
+    trials: int
+    graph_type: str
+    delta_gamma: float
+    delta_beta: float
```

Comment:
Here we should have a more minimal set of metrics. Most of them are already in the input parameters of the benchmark or in the intermediate data, they don't need to be in the results.

The way we think of the BenchmarkResult object is basically whatever number you would put in a table like this one from your paper, and nothing else. 

<img width="438" height="547" alt="Screenshot 2025-08-15 at 12 10 14" src="https://github.com/user-attachments/assets/a46756e9-8717-45c8-8cc6-569dce61baf5" />